File: 9-PalindromeNumber/9-PalindromeNumber.js
// Last updated: 8/16/2026, 9:24:44 PM
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
 
let  num = x;
let revrse = 0;

while(x > 0){
let digit = x%10;
 revrse = revrse*10 + digit;
x = Math.floor(x/10)
}

if(num == revrse){
return true;
}else{
  return false;
}

};

File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 8/16/2026, 9:24:38 PM
class Solution {
public:
    bool isPalindrome(int x) {

        // negative numbers are not palindrome
        // numbers ending with 0 cannot be palindrome
        // except 0 itself
        if(x < 0 || (x % 10 == 0 && x != 0)){
            return false;
        }

        int rev = 0;

        while(x > rev){

            rev = rev * 10 + x % 10;

            x = x / 10;
        }

        // even digits: x == rev
        // odd digits: x == rev/10

        return (x == rev || x == rev / 10);
    }
};